components/category/Category.js

```diff
@@ -25,52 +23,84 @@ export default function Category({ category, listId }) {
     )
       return;
 
-    updateListItemIndex(destination.index, source.index);
+    updateListItemIndex(destination, source);
   };
 
+  //activeListItems
+  const activeListItems = listItems.filter(
+    (listItem) => listItem.categoryId === category.id && !listItem.disabled
+  );
+  //inactiveListItems
+  const inactiveListItems = listItems.filter(
+    (listItem) => listItem.categoryId === category.id && listItem.disabled
+  );
+
   return (
-    <article>
+    <CategoryWrapper disabled={hasDisabledItems}>
       <CategoryHeadline
         id={category.id}
         extended={isExtended}
         handleClick={() => setIsExtended((oldIsExtended) => !oldIsExtended)}
         listId={listId}
+        disabled={hasDisabledItems}
       >
         {category.name}
       </CategoryHeadline>
       {isExtended && (
         <DragDropContext onDragEnd={handleOnDragEnd}>
-          <Droppable droppableId={category.id}>
+          <Droppable
+            droppableId={
+              hasDisabledItems ? "disabled" + category.id : category.id
```

Comment:
modifying the id here is pretty hacky and changes the semantic meaning and information the id contains. A better way would be to give your  updateListItemIndex function another (boolean) parameter, for example like this:
```
updateListItemIndex: (destination, source, onlyDisabledItems) 
```